### DISCUSSION TITLE
LineChart not showing axis Labels

### DISCUSSION THREAD

**Discussion by DSanjulian** [patrykandpatrick/vico#839] [Questions]
### Question

Hi, I am new to vico (and GitHub for that matter, sorry for the format as well), so sorry for the supernoob question, but I can't get a lineChart to show the y-axis labels.
Here's the Chart:
` fun LineChartComponent(weights: List<Pair<Float, Float>>) {
    val chartEntries = List(weights.size) {entryOf(it.toFloat(), weights[it].second)}
    val chartEntryModelProducer = ChartEntryModelProducer(chartEntries)
    Chart(
        chart = lineChart(),
        chartModelProducer = chartEntryModelProducer,
        startAxis = rememberStartAxis(
            label = axisLabelComponent(textSize = 15.sp),
            guideline = axisGuidelineComponent(thickness = 1.dp),
            itemPlacer = remember { AxisItemPlacer.Vertical.default() }),
        bottomAxis = rememberBottomAxis(guideline = axisGuidelineComponent(thickness = 0.dp)),
        chartScrollSpec = rememberChartScrollSpec(isScrollEnabled = false)
    )
}`

Thank you!

### Vico version(s)

1.15.0

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hi @DSanjulian, there's the same problem with `setEntries` as I mentioned in #841. Also, you should `remember` the `ChartEntryModelProducer`, as you did in #841, or move the producer to a `ViewModel`.

Here's a snippet with the fixes applied.
```kt
fun LineChartComponent(weights: List<Pair<Float, Float>>) {
    val chartEntryModelProducer = remember { ChartEntryModelProducer() }
    LaunchedEffect(weights) {
        chartEntryModelProducer.setEntries(weights.map { entryOf(it.first, it.second) })
    }
    Chart(
        chart = lineChart(),
        chartModelProducer = chartEntryModelProducer,
        startAxis =
            rememberStartAxis(
                label = axisLabelComponent(textSize = 15.sp),
                guideline = axisGuidelineComponent(thickness = 1.dp),
                itemPlacer = remember { AxisItemPlacer.Vertical.default() }
            ),
        bottomAxis = rememberBottomAxis(guideline = null),
        chartScrollSpec = rememberChartScrollSpec(isScrollEnabled = false)
    )
}
```

This should work. If you still encounter issues, please share the entire example along with the data used for the chart.

**Reply by zainfur** [reply]
`rememberBottomAxis ` and `rememberStartAxis ` are unresolved for - I'm using the latest SDK 35

**Reply by Gowsky** [reply]
Hello, @zainfur. This discussion concerns Vico 1.15.0. The functions have been renamed in Vico 2. Please take a look at the guide.

**Reply by stawilL20** [reply]
The guide still has rememberStartAxis() and rememberBottomAxis() here: https://www.patrykandpatrick.com/vico/guide/stable/compose/cartesian-charts/starter-examples

It hasn't been updated to reflect that they should now be VerticalAxis.rememberStart() and HorizontalAxis.rememberBottom()

**Reply by patrickmichalik** [reply]
Thanks for the heads-up, @stawilL20! We’ve corrected this.
